plugin/checkout_handlers.go

```diff
@@ -0,0 +1,66 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+var CurrentlyCheckedOut = errors.New("currently checked out")
+
+type CheckOut struct {
+	BorrowerEntityID    string        `json:"borrower_entity_id"`
+	BorrowerClientToken string        `json:"borrower_client_token"`
+	LendingPeriod       time.Duration `json:"lending_period"`
+	Due                 time.Time     `json:"due"`
+}
+
+type CheckOutHandler interface {
+	CheckOut(ctx context.Context, storage logical.Storage, serviceAccountName string, checkOut *CheckOut) error
+	CheckIn(ctx context.Context, storage logical.Storage, serviceAccountName string) error
+	Delete(ctx context.Context, storage logical.Storage, serviceAccountName string) error
+	Status(ctx context.Context, storage logical.Storage, serviceAccountName string) (*CheckOut, error)
+}
+
+type StorageHandler struct{}
+
+func (h *StorageHandler) CheckOut(ctx context.Context, storage logical.Storage, serviceAccountName string, checkOut *CheckOut) error {
+	// Check if the service account is currently checked out.
```

Comment:
Test for `checkOut == nil`

Reply:
Well, the checkOut should never be nil or the caller is breaking the contract laid out by the signature. Also, calls will flow from the OverdueWatcher -> ServiceAccountLocker -> PasswordHandler -> StorageHandler. So, if I'm going to validate the inputs, I'd prefer to do that at the outermost layer, rather than at the innermost layer or every layer. 

So, I think we could accomplish this alternatively by:
1. Documenting on the interface that all inputs must be populated
2. Making an outermost object (before the OverdueWatcher) that's an InputValidator and makes sure everything is populated
3. Both 1 and 2

What do you think?

Reply:
I'll just go ahead and do both of those things, because I think that solves the concern.

Reply:
also is "checkout"  different from "check out" here? At first I thought the method should be `Checkout` but now I'm questioning my grip on the English language 🤔

Reply:
Hm, well, I am required to export the method in order to meet the interface. I think I can find common ground, then, by making a private method or two that's like `validateInputs` and then I can reuse it on all the methods that will be exported. I'll take that route.

Reply:
Regarding "check out", I've seen it [reflected as two words](https://www.google.com/search?q=check+out&oq=check+out&aqs=chrome.0.0l4j69i61l2.912j0j7&sourceid=chrome&ie=UTF-8), and also with a hyphen ("check-out"). I think because of that, `CheckOut` and `checkOut` should both get the message across okay.

Reply:
Marking this as resolved because I've pushed up an input validation method with unit test coverage.